scrapeNews/scrapeNews/pipelines.py

```diff
@@ -78,93 +100,51 @@ def open_spider(self, spider):
             loggerError.error(Error)
 
 
-    def close_spider(self, spider):
+    def closeConnection(self, spiderName):
+        # closes connection with postgreSQL using pyscopg2
         try:
             self.cursor.close()
             self.connection.close()
             if self.recordedArticles > 0:
-                loggerInfo.info(str(self.recordedArticles) + " record(s) were added by " + spider.name + " at")
+                loggerInfo.info(str(self.recordedArticles) + " record(s) were added by " + spiderName + " at")
         except Exception as Error:
             loggerError.error(Error)
 
 
-    def process_item(self, item, spider):
+    def insertIntoNewsTable(self, item, spider):
+        # Insert item into NEWS_TABLE after all the processing.
         try:
-            postgresQuery = "SELECT link from " + os.environ['NEWS_TABLE'] + " where link= %s"
-            self.cursor.execute(postgresQuery, (item.get('link'),))
-            if not self.cursor.fetchall():
-                processedDate = self.process_date(item.get('newsDate'),item.get('link'), spider.name)
-                try:
-                    postgresQuery = "INSERT INTO " + os.environ['NEWS_TABLE'] + " (title, content, image, link, newsDate, site_id, datescraped) VALUES (%s, %s, %s, %s, %s, %s, NOW())"
-                    self.cursor.execute(postgresQuery,
-                        (item.get('title'),
-                        item.get('content'),
-                        item.get('image'),
-                        item.get('link'),
-                        processedDate,
-                        item.get('source')))
-                    self.recordedArticles += 1
-                except Exception as Error:
-                    loggerError.error(str(Error) + " occured at " + str(item.get('link')))
-                finally:
-                    return item
+            if (self.connection.status != 1):
+                self.openConnection()
+            postgresQuery = "INSERT INTO " + os.environ['NEWS_TABLE'] + " (title, content, image, link, newsDate, site_id, datescraped) VALUES (%s, %s, %s, %s, %s, %s, NOW())"
+            processedDate = str(parser.parse(item.get('newsDate'), ignoretz=False))
+            self.cursor.execute(postgresQuery,
```

Comment:
Implement None or Null Checks before executing this.

Reply:
None or Null will automatically be rejected by the database, that's why i removed it!